commit 682542646d32db974f260c408cb78e7a7f2314c1
Author: bertin <macportsuser@macports.org>
Date:   Fri Oct 10 19:16:21 2025 +0200

    /opt/local/linux-ports/audio/audacity/files/patch-scrubui-issue-9585.diff

diff --git a/src/ProjectWindow.cpp b/src/ProjectWindow.cpp
index b0de71749..cabfd492d 100644
--- a/src/ProjectWindow.cpp
+++ b/src/ProjectWindow.cpp
@@ -615,6 +615,11 @@ ProjectWindow::ProjectWindow(wxWindow * parent, wxWindowID id,
 
 ProjectWindow::~ProjectWindow()
 {
+   auto removed = PopEventHandler();
+   while (removed) {
+//       fprintf(stderr, "%s: popped event handler %p for window %p\n", __PRETTY_FUNCTION__, removed, this);
+      removed = PopEventHandler();
+   }
    // Tool manager gives us capture sometimes
    if(HasCapture())
       ReleaseMouse();
diff --git a/src/tracks/ui/ScrubUI.cpp b/src/tracks/ui/ScrubUI.cpp
index b09e4de59..d4f947692 100644
--- a/src/tracks/ui/ScrubUI.cpp
+++ b/src/tracks/ui/ScrubUI.cpp
@@ -227,12 +227,17 @@ struct ScrubForwarder
 
    ~ScrubForwarder()
    {
-      if ( mWindow )
-         mWindow->PopEventHandler();
+//       fprintf(stderr, "%s(%p): mWindow.get()=%p\n", __PRETTY_FUNCTION__, this,
+//               mWindow.get());
+//       fflush(stderr);
+      if ( mWindow ) {
+         /*const auto removed = */mWindow->RemoveEventHandler(this);
+//          fprintf(stderr, "\t%s an event handler!\n", removed? "Removed us as " : "we were no longer");
+      }
    }
 
    AudacityProject &mProject;
-   wxWindowPtr<wxWindow> mWindow;
+   wxWeakRef<wxWindow> mWindow;
    wxWeakRef<AdornedRulerPanel> mRuler;
    std::weak_ptr<Scrubber> mScrubber;
 
@@ -289,6 +294,10 @@ END_EVENT_TABLE()
 
 static const AudacityProject::AttachedObjects::RegisteredFactory sForwarderKey{
    []( AudacityProject &parent ){
-      return std::make_shared< ScrubForwarder >( parent );
+      auto ret = std::make_shared< ScrubForwarder >( parent );
+//       fprintf(stderr, "%s: made shared ScrubForwarder %p for parent %p, window %p\n",
+//               __PRETTY_FUNCTION__, ret.get(), &parent, &ProjectWindow::Get(parent));
+//       fflush(stderr);
+      return ret;
    }
 };
